cpu/esp32/esp-eth/esp_eth_netdev.c

```diff
@@ -107,6 +105,7 @@ static esp_err_t IRAM_ATTR _eth_input_callback(void *buffer, uint16_t len, void
     if (_esp_eth_dev.event == SYSTEM_EVENT_MAX ||
         _esp_eth_dev.event == SYSTEM_EVENT_ETH_RX_DONE) {
         memcpy(_esp_eth_dev.rx_buf, buffer, len);
+        free(buffer);
```

Comment:
Where was that buffer allocated?
Not in this file - better add a comment.

Reply:
This buffer is allocated in EMAC driver of ESP-IDF and has to be freed by the consumer.